### PR TITLE
Fix building on AppVeyor

### PR DESCRIPTION
**Pull request by pali**
Fix configure dependences, test with more Perl versions and set
environment variables which are also on Travis.

### CODE REVIEW

File: Makefile.PL [hunk 1]
```diff
@@ -525,12 +525,23 @@ USAGE
 }
 
 
+sub Execute {
+  my ($command) = @_;
+  # Silence "Can't exec" warnings
+  local $SIG{__WARN__} = sub {};
+  my $output = `$command 2>&1`;
+  # Detect command not found on Windows
+  $output = undef if $^O eq 'MSWin32' and defined $output and $output =~ /is not recognized as/;
```
Comment (choroba) [inline Makefile.PL]: Are all of `$?`, `$^E`, and `${^CHILD_ERROR_NATIVE}` ignored?

Comment (pali) [reply Makefile.PL]: `$?` is exit code, so not suitable
`$^E` is value of the last `GetLastError()` WINAPI call, nothing interesting
`${^CHILD_ERROR_NATIVE}` seems like extended exit code

As `$output` contains something, it means that process started and successfully finished. Documentation for backtick operator says: `A string which is (possibly) interpolated and then executed as a system command with /bin/sh or its equivalent.`. So I guess it spawns `cmd.exe /c $command`, cmd.exe prints error message that `$command` was not recognized and exit...

Comment (choroba) [reply Makefile.PL]: I fear that the error message can be localized in MSWin :fearful:

Comment (pali) [reply Makefile.PL]: Looks like that `system { $command } $command` does not work when `$command` is batch file without specified extension. Perl always throw exec warning `Can't spawn "cmd.exe": No such file or directory`. Return value of `system` function, `$?` and `${^CHILD_ERROR_NATIVE}` is 65280, `$!` and `$^E` is 2.

`system($command)` returns for non-existing command value 256, `$?` and `${^CHILD_ERROR_NATIVE}` is set to 256, `$!` and `$^E` is set to 2.

`system($command)` for existing command returns value `$exitcode << 8`, also `$?` and `${^CHILD_ERROR_NATIVE}` are set to this value. `$!` and `$^E` are set to 2 too!

So it is not possible to distinguish between command does not exist and command returned exit code 1.

Comment (pali) [reply Makefile.PL]: I changed this function to use `MM->maybe_command`. Seems it does right thing.

File: Makefile.PL [hunk 2]
```diff
@@ -525,13 +525,23 @@ USAGE
...
+
+
 my %ParsedCache;
 sub ParseMySQLConfig {
   my ($opt, $param) = @_;
   return $ParsedCache{$param} if exists $ParsedCache{$param};
   return undef unless $opt->{'mysql_config'} and $opt->{'mysql_config'} ne 'skip';
   my $output = `$opt->{'mysql_config'} --$param 2>&1`;
-  if (defined $output and $output !~ /Options:|Usage:|unrecognized option/) {
+  if ($? >= 0 and !($? & 127) and defined $output and $output !~ /Options:|Usage:|unrecognized option/) {
```
Comment (choroba) [inline Makefile.PL]: `$? == 0`

**Pull request merged**